src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java

```diff
@@ -462,6 +468,126 @@ public void testConnectionPoolGetTwice() throws SQLException {
         }
     }
 
+    /**
+     * Test connection properties: CONCAT_NULL_YIELDS_NULL with SQLServerXADataSource for new connection and pooled connection
+     * @throws SQLException
+     */
+    @Test
+    public void testConcatNullYieldsNull() throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setConcatNullYieldsNull("OFF");
+        int expectedResultForNewConnection = 0;
+        // Server default is CONCAT_NULL_YIELDS_NULL = ON  
+        int expectedResultForPooledConnection = 1;

```

Comment:
When we set concat null yields null to OFF, the expected result == 0 for all connections, even pooled connections. If that's not happening, we need to figure out what's not working correctly in the implementation so it can be fixed.